server/src/main/java/org/apache/druid/server/coordinator/duty/CompactSegments.java

```diff
@@ -277,7 +300,8 @@ private CoordinatorStats doRun(
         );
         LOG.infoSegments(segmentsToCompact, "Compacting segments");
         // Count the compaction task itself + its sub tasks
-        numSubmittedTasks += findNumMaxConcurrentSubTasks(config.getTuningConfig()) + 1;
+        numSubmittedTasks++;
+        numCompactionTasksAndSubtasks += findMaxNumTaskSlotsUsedByOneCompactionTask(config.getTuningConfig()) + 1;
```

Comment:
I might be missing something....but in a non parallel mode, wouldn't this still count as 2 instead of 1?

Reply:
^ Are there tests for this?

Reply:
Oops, good catch. It was always counting an extra slot regardless of the mode of the compaction task. Added tests.